src/server/api/routers/application.ts

```diff
@@ -1,10 +1,27 @@
-import { z } from "zod";
+import { TRPCError } from "@trpc/server";
+import { eq } from "drizzle-orm";
 
 import { createTRPCRouter, protectedProcedure } from "~/server/api/trpc";
+import { applications } from "~/server/db/schema";
 
 export const applicationRouter = createTRPCRouter({
   get: protectedProcedure.query(({ ctx }) => {
-    // TODO: finish body
+    const userId = ctx.session.user.id;
+    ctx.db.select().from(applications).where(eq(applications.userId, userId)).limit(1)
+      .then((application) => {
+        if (!application.length) {
+          throw new TRPCError({
+            code: "NOT_FOUND",
+            message: "Application not found",
+          });
+        }
+        return application[0];
+      }).catch((error) => {
+        throw new TRPCError({
+          code: "INTERNAL_SERVER_ERROR",
+          message: "Failed to fetch application: " + JSON.stringify(error),
+        });
```

Comment:
So I just approved, but realized that this will always wrap a 'NOT_FOUND' error in the 'INTERNAL_SERVER_ERROR'... I think this is fine though so I'm going to keep this approved, just thought I'd mention this in case that's not what you think we should do.